packages/graphql-codegen-cli/tests/generate-and-save.spec.ts

```diff
@@ -254,4 +254,103 @@ describe('generate-and-save', () => {
     // makes sure it doesn't write a new file
     expect(writeSpy).toHaveBeenCalled();
   });
+
+  describe('Syntax errors when loading pointers', () => {
+    const originalConsole = { ...console };
+    const originalNodeEnv = process.env.NODE_ENV;
+
+    let consoleErrorMock;
+
+    beforeEach(() => {
+      // Mock common console functions to avoid noise in the terminal
+      global.console.log = jest.fn();
+      global.console.warn = jest.fn();
+      global.console.error = jest.fn();
+
+      // By default, the NODE_ENV is set to 'test', and this is used to silent console errors.
+      // For these tests below, we want to see what's being logged out to console errors.
+      process.env.NODE_ENV = 'not_test_so_error';
+
+      consoleErrorMock = jest.mocked(global.console.error);
+    });
+
+    afterEach(() => {
+      global.console = originalConsole;
+      process.env.NODE_ENV = originalNodeEnv;
+    });
+
+    test('Schema syntax error - should print native GraphQLError for', async () => {
+      try {
+        await generate(
+          {
+            verbose: true,
+            schema: './tests/test-files/schema-dir/error-schema.graphql',
+            generates: {
+              'src/test.ts': {
+                plugins: ['typescript'],
+              },
+            },
+          },
+          false
+        );
+      } catch {
+        expect(consoleErrorMock.mock.calls[0][0]).toBeSimilarStringTo(
+          '[FAILED] Failed to load schema from ./tests/test-files/schema-dir/error-schema.graphql:'
+        );
+        expect(consoleErrorMock.mock.calls[0][0]).toBeSimilarStringTo(
+          '[FAILED] Syntax Error: Expected Name, found "!".'
+        );
+        // We can only use partial file path to the error file, because the error contains absolute path on the host machine
+        expect(consoleErrorMock.mock.calls[0][0]).toBeSimilarStringTo(
+          '/tests/test-files/schema-dir/error-schema.graphql:2:15'
+        );
+        expect(consoleErrorMock.mock.calls[0][0]).toBeSimilarStringTo(`
+          [FAILED] 1 | type Query {
+          [FAILED] 2 |   foo: String!!
+          [FAILED]   |               ^
+          [FAILED] 3 | }
```

Comment:
GraphQLError Formatting looks correct in `--verbose` scenario, if we use the Listr default renderer, it may look like this in the terminal:

<img width="490" alt="Screenshot 2025-04-04 at 12 17 24 am" src="https://github.com/user-attachments/assets/18317922-c2cc-4a79-a5f0-f1e74fd7199f" />

It looks like the pad left doesn't work on the line with the pointer `^`. This is a bit annoying but we can't easily fix this unless we fix `graphql-js`'s error logging (high effort, not much impact right now IMO)

Reply:
nice!!!!!